protostar/argument_parser/arg_type.py

```diff
@@ -1,30 +1,24 @@
 import re
 from pathlib import Path
 from re import Pattern
-from typing import Any, Callable, Literal
-
-ArgTypeName = Literal[
-    "str",
-    "directory",
-    "path",
-    "bool",
-    "regexp",
-    "int",
-]
-
-
-def map_type_name_to_parser(argument_type: ArgTypeName) -> Callable[[str], Any]:
-    type_name_to_parser_mapping: dict[ArgTypeName, Callable[[str], Any]] = {
-        "str": str,
-        "bool": parse_bool_arg_type,
-        "int": int,
-        "directory": parse_directory_arg_type,
-        "regexp": re.compile,
-        "path": Path,
-    }
-    if argument_type in type_name_to_parser_mapping:
-        return type_name_to_parser_mapping[argument_type]
-    assert False, f"Unknown argument type {argument_type}"
+from typing import Any, Callable
+
+from .types import ParserFactoryProtocol, ArgTypeName
+
+
+class StandardParserFactory(ParserFactoryProtocol[ArgTypeName]):
```

Comment:
It is not obvious what is the purpose of this class

Reply:
I do not see why this is a factory instead of just `Parser`, It would remove one layer of abstraction.